app/controllers/api/v1/projects_controller.rb

```diff
@@ -1,11 +1,63 @@
-class Api::V1::ProjectsController < ApplicationController
+include Rails.application.routes.url_helpers
+
+
+class Api::V1::ProjectsController < Api::BaseController
+  def show
+    project = Project.find_by(id: params[:id])
+    unless project
+      render json: { status: "Not Found", data: "Project not found" }, status: :not_found
+      return
+    end
+    owner = project.memberships.owner.first&.user
+    unless check_user_is_public(owner)
+      return
+    end
+
+    render json: { status: "Success", data: project_data(project) }, status: :ok
+  end
+
+
   def index
-    @pagy, @projects = pagy(Project.where(deleted_at: nil), items: 100)
+    if @current_user&.admin? || @current_user&.super_admin?
+      projects = Project.all
```

Comment:
We shouldn't Project.all here. This combined with the includes could be generating a join with 1mil+ records across 5 (project,membership,user,vote,post/post::devlog) tables